crates/matrix-sdk-crypto/src/olm/group_sessions/mod.rs

```diff
@@ -159,20 +159,19 @@ impl From<ToDeviceForwardedRoomKeyEventContent> for ExportedRoomKey {
     }
 }
 
+#[cfg(target_os = "linux")]
```

Comment:
Because substracting a `Duration` from an `Instant` panics if we're not on linux, this line: https://github.com/matrix-org/matrix-rust-sdk/blob/8ab5df490a326f91f692936cee8bb8bbe66df69c/crates/matrix-sdk-crypto/src/olm/group_sessions/mod.rs#L204.

Reply:
hmm... that's odd, because we are using [`instant::Instant` here avoid exactly that problem](https://github.com/sebcrozet/instant).

Reply:
`instant::Instant` is just `std::time::Instant` on native platforms: https://github.com/sebcrozet/instant/blob/master/src/native.rs

`std::time::Instant` hints at this over here: https://doc.rust-lang.org/std/time/struct.Instant.html#os-specific-behaviors.

Reply:
looks like the issue is with the [platform underneath not supporting `performance.now()`](https://github.com/sebcrozet/instant#using-instant-for-a-wasm-platform-where-performancenow-is-not-available), switching the [instant features from `now` to `inaccurate` solves that issue](https://github.com/matrix-org/matrix-rust-sdk/pull/414/commits/2ee7a95fc483140605e615a9ba5b6d2760e97974#diff-b104276a39ce1fe794b39d471d310304b00e1b77771630f1e16dff7f7d54f7f6R29).

Reply:
Err, I mean, this will panic on macOS and Windows likely as well. It wasn't a WASM specific issue.

Reply:
[indeed](https://github.com/matrix-org/matrix-rust-sdk/runs/4384741432?check_suite_focus=true#step:6:404). fascinating. didn't know that duration subtraction panics on macos...